src/vaults.lua

```diff
@@ -6,6 +6,9 @@ local balances = require("balances")
 local utils = require("utils")
 local constants = require("constants")
 
+--- @type Timestamp|nil
+NextPruneTimestamp = NextPruneTimestamp or 0
```

Comment:
will this be the global timestamp for all pruning activity? if so, consider moving to main.lua

Reply:
also may want to consider if we pass this down to prune functions rather than referencing the global timestamp everywhere.

Reply:
a056f58

Reply:
> also may want to consider if we pass this down to prune functions rather than referencing the global timestamp everywhere.

The global is the lightest lift at the moment. Otherwise I'll have to collect all the outputs of every usage of the functions that might mutate the next timestamp value and apply that to a different global that lives somewhere else. Need to globals to survive evals.